pkg/remediator/watch/manager.go

```diff
@@ -217,22 +218,21 @@ func (m *Manager) UpdateWatches(ctx context.Context, gvkMap map[schema.GroupVers
 	var errs status.MultiError
 	for gvk := range gvkMap {
 		// Skip watchers that are already started
-		if _, isWatched := m.watcherMap[gvk]; isWatched {
+		if w, isWatched := m.watcherMap[gvk]; isWatched {
+			w.SetLatestCommit(commit)
 			continue
 		}
 		// Only start watcher if the resource exists.
 		// All resources should exist and be established by now. If not, error.
 		if _, err := m.mapper.RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
-			switch {
-			case meta.IsNoMatchError(err):
-				errs = status.Append(errs, syncerclient.ConflictWatchResourceDoesNotExist(err, gvk))
-			default:
-				errs = status.Append(errs, status.APIServerErrorWrap(err))
+			if meta.IsNoMatchError(err) {
+				metrics.RecordResourceConflict(ctx, commit)
```

Comment:
Probably still want to wrap this as a ConflictWatchResourceDoesNotExist so it gets the more descriptive error message. We can just also immediately record the metric.

Reply:
Done